File: app/src/main/java/com/fafu/app/dfb/mvp/base/i/IView.java
package com.fafu.app.dfb.mvp.base.i;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

public interface IView {

    /**
     * 显示加载
     */
    default void showLoading() {}

    /**
     * 隐藏加载
     */
    default void hideLoading() {}

    /**
     * 显示消息
     */
    void showMessage(String msg);

    void openActivity(Intent intent);

    Context getContext();

    Activity getActivity();

    void killSelf();
}

File: app/src/main/java/com/fafu/app/dfb/mvp/base/i/IView.kt
package com.fafu.app.dfb.mvp.base.i

import android.app.Activity
import android.content.Context
import android.content.Intent

interface IView {
    /**
     * 显示加载
     */
    fun showLoading() {}

    /**
     * 隐藏加载
     */
    fun hideLoading() {}

    /**
     * 显示消息
     */
    fun showMessage(msg: String)

    fun openActivity(intent: Intent)

    fun getContext(): Context

    fun getActivity(): Activity

    fun killSelf()

}